domain/consensus/processes/consensusstatemanager/resolve_virtual_test.go

```diff
@@ -212,3 +215,106 @@ func TestAddGenesisChildAfterTwoResolveVirtualCalls(t *testing.T) {
 		}
 	})
 }
+
+func TestResolveVirtualMess(t *testing.T) {
```

Comment:
Please add explanation of what this mess is. Also maybe rename the test to something more "formal" than "mess"